docs/source/references/api_spec.md

```diff
@@ -70,6 +70,31 @@ Server sends `Event` and expects `EventReceived` after each, before sending the
 - 200 OK - Metrics Calculated Successfully. Body: `Metrics`
 - 500 Internal Server Error - Failed to get metrics
 
+### Configuration
+
+**Protocol**: HTTP
+
+**Encoding**: Json
+
+**Endpoint**: `/configuration`
+
+**Method**: `GET` or `PUT`
+
+**Expects** on `GET`:
+Query parameter "docs" can be set to boolean (indecates whether should return documentation or field value)
+Request body is json with field "field", which is array of strings.
+
+**Responses** on `GET`:
+- 200 OK - Field was found and either doc or value is returned in json body.
```

Comment:
What will happen if such field not exists? It will return an empty json `{}` or 404 status code?

Reply:
Actually it is a mistake here, will change doc. 400 bad request is returned now actually :)

Reply:
I think 400 status code is not suitable here due request itself is correct - server was able to parse it and no syntax error was met. https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400 
I think better to return an empty json

Reply:
Ok then. There are actually 2 class of errors here. One if field is not found and we can return 404 here, the other one happends when client sends bad value which can't be deserialized into field.

What status code we should return when?

Reply:
I think
- if configuration field not found then  return 200 status code with an empty json body `{}`
- if input value in request cannot be deserialized then return 403 status code

Reply:
In current state it is easier to return just 404 with error description for unknown field.

Reply:
I agree. 400 is better in case of invalid data type (for example, providing a string in a numeric field)